### PR TITLE
Fix total size count for data scroller

### LINKED ISSUE #6087: DataScroller: Lazy loading losing track of items loaded
Running clean version 8.   2nd chunk replaces item association with first chunk. The issue is the exact same issue as #5354 which was closed with a pull request, but I noticed that the changes were reverted by a subsequent change three months later (I think #5612).

As soon as I work through the issues I'm having building my own version of 9.0-SNAPSHOT I'll retest it there, but the code looks unchanged.

**Environment:**
 - PF Version: _8.0_
 - JSF : jboss-jsf-api_2.3_spec-3.0.0.Final.jar (WildFly 11)
 - Affected browsers: _e.g. Chrome, iOS, ALL_

### LINKED ISSUE #11019: DataScroller: with lazyloading only load one time
### Describe the bug

When I try to use a dataScroller with lazy loading, the data load only once.  

### Reproducer

I've made a reproducer : https://github.com/Samuel-StO/primefaces-datascroller-issue

In the reproducer, I have a list of 100 cities that I'm trying to display in a lazy loading datascroller. 
I've set the number of elements per chunk to 25. 

The behavior I'm observing is that at first there are 25 elements, then if I scroll it loads the next 25. But if you keep scrolling, it won't load the next 25. 
In fact, it isn't capable of loading more data than once, even if there's more data to load. 

The video shows problematic behavior :
https://github.com/primefaces/primefaces/assets/86346301/c9a18e2c-85b3-4355-80a5-d9b90c44c515

### Expected behavior

The desired behavior is for the datascroller to continue loading data when you scroll.

### PrimeFaces edition

Community

### PrimeFaces version

13.0.3

### Theme

_No response_

### JSF implementation

Mojarra

### JSF version

2.3

### Java version

8

### Browser(s)

_No response_

### PR DESCRIPTION
**Pull request by Samuel-StO**
Fix #11019
Fix #6087 

Use the count method for data scroller when we are in lazy

### CODE REVIEW

File: primefaces/src/main/java/org/primefaces/component/datascroller/DataScrollerRenderer.java [hunk 1]
```diff
@@ -192,6 +192,16 @@ protected void encodeScript(FacesContext context, DataScroller ds, int chunkSize
 
         wb.finish();
     }
+    
+    private static int getTotalSize(DataScroller ds) {
```
Comment (tandraschko) [inline primefaces/src/main/java/org/primefaces/component/datascroller/DataScrollerRenderer.java]: private static -> protected

File: primefaces/src/main/java/org/primefaces/component/datascroller/DataScrollerRenderer.java [hunk 2]
```diff
@@ -192,6 +192,16 @@ protected void encodeScript(FacesContext context, DataScroller ds, int chunkSize
...
+        if ( ds.isLazy() ) {
```
Comment (tandraschko) [inline primefaces/src/main/java/org/primefaces/component/datascroller/DataScrollerRenderer.java]: our styleguide is `if (ds.isLazy()) {` (remove spaces)

**Comment by tandraschko** [comment]
checkstyle still failing, i wonder if you even built the source on your local machine before?

**Review by melloware — APPROVED** [review]
build is passing now.  LGTM

**Comment by tandraschko** [comment]
looks good for me
@melloware can you apply it and port to master?

**Comment by melloware** [comment]
Yep will do!

**Pull request merged**